library/src/interactor_impl.cxx

```diff
@@ -558,16 +498,92 @@ class interactor_impl::internals
     this->Window.render();
   }
 
+  struct tokenize_exception : public exception
+  {
+    explicit tokenize_exception(const std::string& what = "")
+      : exception(what)
+    {
+    }
+  };
+
+  static std::vector<std::string> Tokenize(std::string_view str)
+  {
+    std::vector<std::string> tokens;
+    std::string token;
+    const auto accumulate = [&](const char& c) { token.push_back(c); };
+    const auto emit = [&]()
+    {
+      if (!token.empty())
+      {
+        tokens.push_back(token);
+        token.clear();
+      }
+    };
+    bool escaped = false;
+    char quoted = '\0';
+    for (const char& c : str)
```

Comment:
if const-ref a string_view is too much, this one too

Reply:
there is no char_view afaik ?

Reply:
`for (char c : str)`

Reply:
right